packages/core/src/plugins.ts

```diff
@@ -0,0 +1,134 @@
+import * as uint8arrays from "uint8arrays"
+
+export type DidKeyPlugin = {
+  prefix: Uint8Array
+  jwtAlg: string
+  didToPublicKey: (did: string) => Uint8Array
+  publicKeyToDid: (pubkey: Uint8Array) => string
+  verifySignature: (publicKey: Uint8Array, data: Uint8Array, sig: Uint8Array) => Promise<boolean>
+}
+
+export type DidMethodPlugin = {
+  isMatch: (method: string, did: string) => boolean
+  checkJwtAlg: (did: string, jwtAlg: string) => boolean
+  verifySignature: (did: string, data: Uint8Array, sig: Uint8Array) => Promise<boolean>
+}
+
+export type Plugins = {
+  keys: DidKeyPlugin[]
+  methods: DidMethodPlugin[]
+}
```

Comment:
I wasn't sure if there was more complex logic around some DID methods that you might want to switch on.

Let's suppose we didn't have the separate plugin system for `did:key` and were instead doing it through the `DidMethodPlugin`. You may want to switch not only on the method, but also on the first few chars of the DID. I was basically trying to future proof it in the event of some DID method with strange switching in the resolution logic.

Going back and thinking it through again though, I think it probably just makes more sense to push that into the plugin logic instead of trying to do it with `isMatch` 👍

Reply:
Ah, I see. Now I understand.
Either way it seems fine to me.